### ISSUE TITLE
New rule suggestion: detect modified const

### ISSUE THREAD

**Issue by AsaAyers** [eslint/eslint#3191]
`prefer-const` can track when I declare a variable that is never modified. It would be nice to have the inverse to detect if I messed up and declared something a `const` but it really needs to be a `let`.

Today I didn't pay close enough attention and I converted a `let { a, b, c } = ...` to a `const { a, b, c } = ...` because `prefer-const` was recommending I convert the first variable, but not all of them.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by AsaAyers** [comment]
1. The use case for the rule - what is it trying to prevent or flag?

```js
const { a, b } = {a: 1, b: 2}
b = 3 // error
```

2. Whether the rule is trying to prevent an error or is purely stylistic

I think it's a syntax error, but I'm not totally sure. I know babel errors when I try to use it, but I don't know how it works when using it in a native runtime.

3. Why you believe this rule is generic enough to be included

modifying a `const` is always an error. It would help to catch it when I write it instead of when I try to compile or run my code.

**Comment by nzakas** [comment]
Already coming in 1.0.0: http://eslint.org/docs/1.0.0/rules/no-const-assign

**Comment by AsaAyers** [comment]
Thanks :) I didn't think to check 1.0
